tests/test_model.py

```diff
@@ -95,6 +96,9 @@ def test_mismatched_train_and_val_loaders(train_val_implicit_data, change_to_mak
     elif change_to_make == 'num_items':
         train.num_items = 1
         val.num_items = 2
+    elif change_to_make == 'negative_sample_type':
+        train.negative_sample_type = 'item'
+        val.negative_sample_type = 'user'
```

Comment:
Can we also add that test we talked about offline which is just the copy of the `test_implicit_model` test in `model.py`. This will unfortunately require creating a billion more movielens + models fixtures (sorry) 😬